backend/dal/company.js

```diff
@@ -0,0 +1,91 @@
+'use strict';
+
+const Promise = require('promise');
+
+module.exports = (connection) => {
+  return {
+    getCompanyById: (id) => {
+      return new Promise((resolve, reject) => {
+        let request = [
+          'SELECT * FROM `company` WHERE id = "' + id + '"'
+        ].join('');
+
+        connection.query(request, (err, response) => {
+          (err || !response.length) ? reject(err) : resolve(response[0]);
+        });
+      });
+    },
+
+    getCompanyByName: (company) => {
+      return new Promise((resolve, reject) => {
+        let request = [
+          'SELECT * FROM `company` WHERE name = "' + company.name + '"'
+        ].join('');
+
+        connection.query(request, (err, response) => {
+          (err || !response.length) ? reject(err) : resolve(response[0]);
+        });
+      });
+    },
+
+    addCompanyForRegister: () => {
```

Comment:
Can we use name like 'add' so wee will call it this way 'DAL.companies.add()'?
Why do we need to add 'ForRegister'? Is it has some sense?